daemon/image_delete.go

```diff
@@ -102,6 +102,10 @@ func (daemon *Daemon) ImageDelete(imageRef string, force, prune bool) ([]types.I
 		// remove that reference.
 		// FIXME: Is this the behavior we want?
 		if len(repoRefs) == 1 {
+			if conflict := daemon.checkImageDeleteConflict(imgID, force); conflict != nil {
```

Comment:
I don't think this is correct because unless `force` is true, `daemon.checkImageDeleteConflict` will always return a soft conflict (because `len(repoRefs) == 1`). I think this needs to check for:

- a hard conflict
- if `force` is false, a stopped container using the image

It probably makes sense to refactor `checkImageDeleteHardConflict` and `checkImageDeleteSoftConflict` a bit to make this easier.